examples/README.md

```diff
@@ -88,3 +120,110 @@ are split for training and the other 11,316 (60,502 images) labels are used for
         ├── 110715681235_0.JPG
         └── ...
 ```
+</p>
+</details>
+
+
+## Usage with custom dataset
+The only difference from the public dataset case is that you have to implement the converter by yourself.
+We expect the `.csv` file in the following format:
+
+Required columns:
+* `label` - integer value indicates the label.
+* `path` - path to sample.
+* `split` - must be one of 2 values: `train` or `validation`.
+* `is_query`, `is_gallery` - have to be `None` where `split == train` and `True`
+or `False` where `split == validation`. Note, that both values can be `True` at
+the same time. Then we will validate every item
+ in the validation set using the "1 vs rest" approach (datasets of this kind are `CARS196` or `CUB`).
+
+Optional columns:
+* `category` - integer value indicates the category if your dataset have one.
+* `category_name` - name of the category.
+* `x_1`, `x_2`, `y_1`, `y_2` - integers, the format is `left`, `right`, `top`, `bot` (`y_1` must be less than `y_2`)
+
+You can check the tables for the public datasets via the [link](https://drive.google.com/drive/folders/12QmUbDrKk7UaYGHreQdz5_nPfXG3klNc?usp=sharing).
+
+
+## How to work with a config?
+We use [Hydra](https://hydra.cc/docs/intro/) as a parser for `.yaml` configs.
+So, you can change whatever you want directly in the config file or override some parameters
+using command line interface:
+```
+python train_cars.py optimizer.args.lr=0.000001 bs_val=128
+```
+
+Instead of changing some parameters in the config file, you may
+also want to completely change the model, loss, optimizer, etc.
+To do this you can check which entities are already available in our registry.
+You can manually inspect `oml.registry` or `oml.configs` or use the function:
+```python
+from oml.registry import show_registry
+show_registry()
+```
+
+## How to use my own implementation of loss, model, augmentations, etc?
+You should put your python object inside the corresponding registry by some key.
+It allows you to access this object in the config file by that key.
+
+You may change the following blocks and to work correctly some of them have to inherit our interfaces:
+* `Transforms`, `Sampler`, `Optimizer`, `Scheduler` - follow the standard PyTorch interfaces.
+* `Model` - have to be child of `IExtractor` (see `oml.interfaces.models`)
```

Comment:
is child correct word?

Reply:
wiki says that child is okay 
https://en.wikipedia.org/wiki/Object-oriented_programming

but I can replace it with `successor`